### PR TITLE
More strict `compilerOptions` for typescript

### PR DESCRIPTION
**Pull request by iTOYS**
Based on #503

### CODE REVIEW

File: src/common/android/androidPlatform.ts [hunk 1]
```diff
@@ -137,7 +142,7 @@ export class AndroidPlatform extends GeneralMobilePlatform {
                     if (apiVersion >= AndroidAPILevel.LOLLIPOP) { // If we support adb reverse
                         return this.adb.reverseAdd(device.id, Packager.DEFAULT_PORT.toString(), this.runOptions.packagerPort);
                     } else {
-                        Log.logWarning(`Device ${device.id} supports only API Level ${apiVersion}. `
+                        return Log.logWarning(`Device ${device.id} supports only API Level ${apiVersion}. `
```
Comment (vladimir-kotikov) [inline src/common/android/androidPlatform.ts]: Though this is correct expression, it's semantically incorrect to "return log warning". Let's change this to a single `return` on a separate line;

File: src/extension/settingsHelper.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export class SettingsHelper {
                 return ConfigurationReader.readString(tsdk);
             }
         }
-        return null;
+        return "";
```
Comment (vladimir-kotikov) [inline src/extension/settingsHelper.ts]: Please don't change the return type of this method - there is a check in `intellisenseHelper` that checks tsdk value against `null` (https://github.com/Microsoft/vscode-react-native/blob/master/src/extension/intellisenseHelper.ts?utf8=%E2%9C%93#L199)

File: tsconfig.json [hunk 3]
```diff
@@ -22,7 +30,6 @@
         "SampleApplication/.vscode",
         "ReactTypings",
         "tools/",
-        "src/test/resources/sampleReactNative022Project/",
-        ".vscode-test"
```
Comment (vladimir-kotikov) [inline tsconfig.json]: I think you still need this line - this dir is created each time after running `npm test`

Comment (iTOYS) [inline tsconfig.json]: https://github.com/Microsoft/vscode-react-native/pull/507/files/37315c47cd153d954c22c9cf543267cd80d95883#diff-e5e546dd2eb0351f813d63d1b39dbc48R26

File: src/common/exponent/exponentHelper.ts [hunk 4]
```diff
@@ -61,11 +61,11 @@ export class ExponentHelper {
                     return showMessage("You need to login to exponent. Please provide username and password to login. If you don't have an account we will create one for you.")
                         .then(() =>
                             promptForInformation("Exponent username", false)
```
Comment (iTOYS) [inline src/common/exponent/exponentHelper.ts]: Without brackets `return` is excess

File: tsconfig.json (hunk 3 above)
Comment (vladimir-kotikov) [inline tsconfig.json]: Oh, missed that line, sorry :)

File: src/common/exponent/exponentHelper.ts (hunk 4 above)
Comment (vladimir-kotikov) [inline src/common/exponent/exponentHelper.ts]: Yep, sorry. Didn't notice that you re using this notation

**Pull request merged**